src/core/or/relay.c

```diff
@@ -2598,8 +2598,14 @@ update_circuit_on_cmux_(circuit_t *circ, cell_direction_t direction,
   /* Now get the cmux */
   cmux = chan->cmux;
 
-  /* Cmux sanity check */
-  if (! circuitmux_is_circuit_attached(cmux, circ)) {
+  /* Now load the circuit map */
+  chanid_circid_muxinfo_t *hashent = NULL;
+  hashent = circuitmux_find_map_entry(cmux, circ);
+  tor_assert(hashent);
+
+  /* Cmux sanity check. We check for hashent == NULL so we call
+   * circuitmux_find_map_entry() less. */
+  if (hashent == NULL) {
```

Comment:
If we are calling tor_assert(hashent) above, it doesn't make sense to test it for non-NULL here.  I think that the tor_assert() above must be wrong, though ... but it makes me worried that this code is not tested enough.